map_model/src/make/turns.rs

```diff
@@ -714,8 +713,8 @@ fn from_pt(pt: Point2d<f64>) -> Pt2D {
 fn is_turn_allowed(turn: &Turn, roads: &Vec<Road>, lanes: &Vec<Lane>) -> bool {
     let l = &lanes[turn.id.src.0];
     let r = &roads[l.parent.0];
-    if let Some(types) = l.get_turn_restrictions(r) {
-        types.contains(&turn.turn_type)
+    if let Some(mut types) = l.get_turn_restrictions(r) {
+        types.any(|turntype| turntype==turn.turn_type)
```

Comment:
This kind of pattern feels a little less readable to me -- it's odd to think about transforming the result of a query.

Reply:
Same here: We can avoid `collect`ing into a temporary data structure just to check that the newly allocated data structure `contains` the element. I agree that `any` is a bit more verbose, but I think we could at least try to avoid this by creating an iterator method `contains` if we consider it useful enough.